src/Misc/PerformanceViewer/dynamicFloat32Array.ts

```diff
@@ -0,0 +1,78 @@
+const growthFactor = 1.5;
+
+/**
+ * A class acting as a dynamic float32array used in the performance viewer
+ */
+export class DynamicFloat32Array {
+    private _view: Float32Array;
+    private _itemLength: number;
+
+    /**
+     * Creates a new DynamicFloat32Array with the desired item capacity.
+     * @param _itemCapacity The initial item capacity you would like to set for the array.
+     */
+     constructor(private _itemCapacity: number) {

```

Comment:
nit: why do you need to store itemCapacity? Is that not _view.length?

Reply:
_view.length may contain elements we haven't initialized as _view.length will get the total capacity we set basically, but the number of actual elements will usually be a lot smaller than this! So this will let us know what the "Usable area" of our array is.

Reply:
Okay yeah I've updated the code and just use the length now LOL, good catch.